src/types.rs

```diff
@@ -244,62 +320,92 @@ macro_rules! new_url_type {
 }
 
 new_url_type![
-    ///
+    /// Base URL of the [Credential] Issuer.
+    IssuerUrl
+    impl {
+        /// Parse a string as a URL, with this URL as the base URL.
+        ///
+        /// See [`Url::parse`].
+        pub fn join(&self, suffix: &str) -> Result<Url, url::ParseError> {
```

Comment:
It's not clear to me why `IssuerUrl` has a different `join` behaviour

Reply:
Most of the other URL newtypes don't need to be modified. The IssuerUrl needs to be joined with a `.well-known/<metadata>` suffix, but `Url::join` replaces the final path element if the URL does not end in a slash, so a custom implementation is needed to avoid that bug.